ui/client/components/graph/node-modal/EdgeDetailsContent.js

```diff
@@ -57,37 +58,34 @@ export default class EdgeDetailsContent extends React.Component {
   }
 
   render() {
-    const { edge, readOnly, updateEdgeProp, showValidation} = this.props
+    const {edge, readOnly, updateEdgeProp, showValidation, showSwitch} = this.props
 
     switch (_.get(edge.edgeType, 'type')) {
       case "SwitchDefault": {
         return this.baseModalContent()
       }
       case "NextSwitch": {
         return this.baseModalContent(
-          <div className="node-row">
-            <div className="node-label">Expression</div>
-            <div className={"node-value"}>
-              <ExpressionSuggest
-                inputProps={{
-                  rows: 1,
-                  cols: 50,
-                  className: "node-input",
-                  value: edge.edgeType.condition.expression,
-                  onValueChange: (newValue) => updateEdgeProp("edgeType.condition.expression", newValue),
-                  language: edge.edgeType.condition.language, readOnly: readOnly}}
-                validators={[notEmptyValidator]}
-                isMarked={this.isMarked("edgeType.condition.expression")}
-                showValidation={showValidation}
-              />
-            </div>
-          </div>
+          <EditableExpression
+            fieldType={"expression"}
+            fieldLabel={"Expression"}
+            renderFieldLabel={this.renderFieldLabel}
+            expressionObj={{expression: edge.edgeType.condition.expression, language: edge.edgeType.condition.language}}
```

Comment:
Maybe you can define expression object earlier?

Reply:
Done